src/App.tsx

```diff
@@ -15,36 +17,159 @@
+  button,
+  sortField,
+  reverse,
+  setSortField,
+  setReverse,
+}) => {
   return (
-    <div className="section content">
-      <div className="buttons">
-        <button type="button" className="button is-info is-light">
-          Sort alphabetically
-        </button>
+    <button
+      type="button"
+      className={cn(
+        'button',
+        { 'is-info': button === 'Sort alphabetically' },
+        { 'is-success': button === 'Sort by length' },
+        { 'is-warning': button === 'Reverse' },
+        { 'is-danger': button === 'Reset' },
+        {
+          'is-light': button === 'Sort alphabetically' && sortField !== 'name',
+        },
+        { 'is-light': button === 'Sort by length' && sortField !== 'length' },
+        { 'is-light': button === 'Reverse' && !reverse },
+        { 'is-light': button === 'Reset' },
+      )}
+      onClick={() => {
+        const value = button;
 
-        <button type="button" className="button is-success is-light">
-          Sort by length
-        </button>
+        switch (value) {
+          case 'Sort alphabetically':
+            setSortField(SortField.Name);
 
-        <button type="button" className="button is-warning is-light">
-          Reverse
-        </button>
+            break;
+          case 'Sort by length':
+            setSortField(SortField.Length);
+            break;
+          case 'Reverse':
+            if (reverse) {
+              setReverse(false);
+            } else {
+              setReverse(true);
+            }
 
-        <button type="button" className="button is-danger is-light">
-          Reset
-        </button>
+            break;
+          case 'Reset':
+            setSortField('');
+            setReverse(false);
+            break;
+
+          default:
+            break;
+        }
+      }}
+    >
+      {button}
+    </button>
+  );
+};
+
+export const App: React.FC = () => {
+  const [sortField, setSortField] = useState('');
+  const [reverse, setReverse] = useState(false);
+  const visibleGoods = getPreparedGoods(goodsFromServer, {
+    sortField,
+    reverse,
+  });
+  const visibleButtons = getPreparedButtons(buttonsFromServer, {
+    sortField,
+    reverse,
+  });
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        {visibleButtons.map(button => (
+          <Button
+            key={button}
+            button={button}
+            sortField={sortField}
+            reverse={reverse}
+            setReverse={setReverse}
+            setSortField={setSortField}
+          />
+        ))}
       </div>
 
       <ul>
-        <ul>
-          <li data-cy="Good">Dumplings</li>
-          <li data-cy="Good">Carrot</li>
-          <li data-cy="Good">Eggs</li>
-          <li data-cy="Good">Ice cream</li>
-          <li data-cy="Good">Apple</li>
-          <li data-cy="Good">...</li>
-        </ul>
+        {visibleGoods.map((good, index) => (
+          <li data-cy="Good" key={index}>
```

Comment:
Using the index as a key in React components can lead to issues, especially when the list can change. It's important to use a unique identifier for each list item to ensure React can efficiently update and manage the list.